main.go

```diff
@@ -432,9 +432,38 @@ func installOption(listAll bool, custBinPath, mirrorURL *string) {
 
 // install when tf file is provided
 func installTFProvidedModule(dir string, custBinPath, mirrorURL *string) {
+	var tfconstraints []string
+	var exactConstraints []string
+
 	logger.Infof("Reading required version from terraform module at %q", dir)
 	module, _ := tfconfig.LoadModule(dir)
-	tfconstraint := module.RequiredCore[0] //we skip duplicated definitions and use only first one
+	requiredVersions := module.RequiredCore
+
+	for key := range requiredVersions {
+		tfconstraint := requiredVersions[key]
+		tfconstraintParts := strings.Fields(tfconstraint)
+
+		if len(tfconstraintParts) > 2 {
+			logger.Fatalf("Invalid version constraint found: %q", tfconstraint)
+			os.Exit(1)
+		} else if len(tfconstraintParts) == 1 {
+			exactConstraints = append(exactConstraints, tfconstraint)
+			tfconstraint = "= " + tfconstraintParts[0]
+		}
+
+		if tfconstraintParts[0] == "=" {
+			exactConstraints = append(exactConstraints, tfconstraint)
+		}
+
+		tfconstraints = append(tfconstraints, tfconstraint)
+	}
+
+	if len(exactConstraints) > 0 && len(tfconstraints) > 1 {
+		logger.Fatalf("Exact constraint (%q) cannot be combined with other conditions", strings.Join(exactConstraints, ", "))
```

Comment:
This came up in a review this morning about testing - for my knowledge, is there a reason for the use of logger.Fatal (it seems to be used quite heavily)? Does this have a value over returning the error upstream (apart from not having to handle the error upstream), because, as a result, it seems that logger.Fatal makes it difficult to add tests(?).

Reply:
Given this function doesn't return anything to upstream (if I'm not missing something), it would be a big piece of re-factoring to streamline that behavior across all codebase and pass errors back to upstream link-by-link to the `main`.
So I simply followed common approach that we're using at the moment, which came from simple switching from `print (+ exit)` to a logger lib.
Hope this answers your question.

Reply:
Yes, it was more of a general question about why this methology is used (in general), not necessarily about these changes in particular :) But yep, all good :)

Reply:
You are right. It would be hard for testing. Back then, we didn't have much test cases. Also, when things fail, we wanted fail fast. 
Ideally I would like to use [context package](https://pkg.go.dev/context) to clean fail and clean up gracefully.

Reply:
I have modified that particular code to return an error an catch it in the invoking function.
That was needed for testing anyways.